### ISSUE TITLE
MvxTableViewSource cant load data

### ISSUE THREAD

**Issue by Nickolas-** [MvvmCross/MvvmCross#4083]
## 🐛 Bug Report

Randomly `MvxTableViewSource` cant load data because tableView in source is null

(MvvmCross.Logging.MvxLog) Exception masked during TableView ReloadData NullReferenceException: Object reference not set to an instance of an object.
	  at MvvmCross.Platforms.Ios.Binding.Views.MvxBaseTableViewSource.ReloadTableData ()  

This happening because of Weak reference on tableView field in `MvxTableViewSource` , if I remove attribute everything work as expected

### Expected behavior

TableView load data

### Reproduction steps

Catch this in release version with link all behaviour and randomly.

### Configuration

**Version:** 7.x

**Platform:** 
- [x] :iphone: iOS
- [ ] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by Cheesebaron** [comment]
I haven't been able to reproduce this. If you have a reproduction project, please provide it.

**Comment by ElteHupkes** [comment]
We also have this happening to people, with `TableView` being `null` in `MvxTableViewSource.CollectionChangedOnCollectionChanged()` (which is what calls `ReloadTableData()`. I unfortunately also haven't seen it happen myself, but I did now receive screenshots from completely empty tables with the custom non-fatal error I added when I saw this one pop up in diagnostics.

How the table view can be garbage collected is beyond me, there's an active view controller with a strong reference to it after all. Will continue to investigate.
